### DISCUSSION TITLE
2023/06/08/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9510-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%AD%97%E7%AC%A6%E7%9A%84%E6%9C%80%E9%95%BF%E5%AD%90%E4%B8%B2/

### DISCUSSION THREAD

**Discussion by giscus** [Zhuanjier/blog#17] [Announcements]
# 2023/06/08/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9510-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%AD%97%E7%AC%A6%E7%9A%84%E6%9C%80%E9%95%BF%E5%AD%90%E4%B8%B2/

¶找到二叉树中最近的右侧节点  原题传送门  给定一个字符串s，请你找出其中不含有重复字符的最长子串的长度。 ¶示例 ¶示例1 输入: s = “abcabcbb” 输出: 3 解释: 因为无重复字符的最长子串是"abc"，所以其长度为 3。 ¶示例2 输入: s = “bbbbb” 输出: 1 解释: 因为无重复字符的最长子串是"b"，所以其长度为 1。

https://blog.zhuanjie.ltd/2023/06/08/leetcode%E5%88%B7%E9%A2%98%E8%AE%B0%E5%BD%9510-%E6%97%A0%E9%87%8D%E5%A4%8D%E5%AD%97%E7%AC%A6%E7%9A%84%E6%9C%80%E9%95%BF%E5%AD%90%E4%B8%B2/

**Comment by RandallAnjie** [comment]
在写这个题目的时候遇到一个小问题，我发现c++中string类型的length()和size()都能获取string类型变量长度，在c++17里面，通过查看源码发现length()方法就是调用了size()方法，所以本人代码中使用size()方法，源码如下：
```c++
_LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_AFTER_CXX17 size_type size() const _NOEXCEPT
        {return __is_long() ? __get_long_size() : __get_short_size();}
    _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_AFTER_CXX17 size_type length() const _NOEXCEPT {return size();}
```
